controllers/vehicle.js

```diff
@@ -0,0 +1,36 @@
+const Vehicle = require('../models/vehicle');
+
+const vehicleController = {
+    createVehicle: async (req, res) => {
+
+        const vehicle = new Vehicle(req.body);
+        if (!vehicle) {
+            return res.status(400).json({ success: false, code: 400, message: 'Error creating Vehicle' })
+        }
+        try {
+            await vehicle.save();
+            res.status(201).json({ success: true, message: "Vehicle created successfully", vehicle });
+        } catch (error) {
+            res.status(400).json({ success: false, code: 400, message: error.message, error: error.errors });
+        }
+    }, 
+    getVehicles: async (req,res) => {
+
+    },
+    getVehicle: (req, res) => {
```

Comment:
Acá es distinto al del usuario, nosotros vamos a tener q recibir el id del vehiculo por parametro de url.
ejemplo -> api/v1/vehicles/145. ese id lo obtenemos gracias al objeto "req" y su atributo params -> req.params
con ese id vamos a usar los metodos q nos facilita mongoose, findOne(id) y si existe lo retornamos, sino retornamos un 404 NOT FOUND. lo vemos el miercoles porque es un caso que no explique